apiserver/facades/agent/uniter/goal-state_test.go

```diff
@@ -0,0 +1,320 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package uniter_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facades/agent/uniter"
+	"github.com/juju/juju/apiserver/params"
+	apiservertesting "github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/juju/testing"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/testing/factory"
+)
+
+// uniterSuite implements common testing suite for all API
+// versions. It's not intended to be used directly or registered as a
+// suite, but embedded.
+type uniterGoalStateSuite struct {
+	testing.JujuConnSuite
+
+	authorizer apiservertesting.FakeAuthorizer
+	resources  *common.Resources
+	uniter     *uniter.UniterAPI
+
+	machine0      *state.Machine
+	machine1      *state.Machine
+	machine2      *state.Machine
+	wordpress     *state.Application
+	wpCharm       *state.Charm
+	mysql         *state.Application
+	wordpressUnit *state.Unit
+	mysqlUnit     *state.Unit
+}
+
+var _ = gc.Suite(&uniterGoalStateSuite{})
+
+func (s *uniterGoalStateSuite) SetUpTest(c *gc.C) {
+	s.JujuConnSuite.SetUpTest(c)
+
+	newFactory := factory.NewFactory(s.State)
+	// Create two machines, two services and add a unit to each service.
```

Comment:
s/service/application

We renamed service to application and not all text was updated, so as we come across old incorrect text we change it as a drive by